metadata/src/test/java/org/apache/kafka/controller/QuorumControllerTest.java

```diff
@@ -688,6 +691,61 @@ public void testNoOpRecordWriteAfterTimeout() throws Throwable {
         }
     }
 
+    @ParameterizedTest
+    @CsvSource(value = {"0, 0", "0, 1", "1, 0", "1, 1"})
+    public void testRegisterBrokerKRaftVersions(short controllerKraftVersion, short brokerKraftVersion) throws Throwable {
```

Comment:
Looks like you are using this to mean the finalized kraft.version. Let's just call it that: `finalizedKraftVersion`.